### PR TITLE
X509 indicator type, layout and fields.

### PR DESCRIPTION
**Pull request by Ni-Knight**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
Moving X509 certificate indicator type from a specific pack into CommonTypes with a new layout and additional fields.

### CODE REVIEW

File: Packs/CommonScripts/ReleaseNotes/1_13_8.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+#### Scripts
+
+##### New: CertificateExtract
+
+- New: Extract fields from a certificate file and return the standard context.
+
+##### New: CertificateReputation
+
+- New: Enrich and calculate the reputation of a certificate indicator. (Available from Cortex XSOAR 6.0.0).
```
Comment (altmannyarden) [inline Packs/CommonScripts/ReleaseNotes/1_13_8.md:10]: how can we determine the certificate reputation? I thought its like MITRE TTP / CVE - where you can choose the default reputation in integration config

File: Packs/CommonTypes/IndicatorFields/indicatorfield-Validity_Not_Before.json [hunk 2]
```diff
@@ -1,30 +1,28 @@
...
+    "associatedToAll": true,
     "unmapped": false,
     "unsearchable": false,
-    "useAsKpi": false,
-    "version": -1,
-    "fromVersion": "6.0.0"
+    "caseInsensitive": true,
+    "sla": 0,
+    "threshold": 72,
+    "fromVersion": "6.10.0"
```
Comment (altmannyarden) [inline Packs/CommonTypes/IndicatorFields/indicatorfield-Validity_Not_Before.json:27]: lower the fromversion back

File: Packs/CommonScripts/Scripts/CertificateReputation/CertificateReputation.py [hunk 3]
```diff
@@ -199,25 +201,31 @@ def dbot_context(value: str, certificate_context: Dict[str, Any]
...
-        if CertificateValidationTag.SELF_SIGNED in tags:
-            if subject_dn is not None and issuer_dn is not None and subject_dn != issuer_dn:
-                comments.append(f'{CertificateValidationTag.INVALID_DISTINGUISHED_NAMES.value}'
-                                ' Self-Signed Certificate with different Issuer DN and Subject DN')
-                tags.append(CertificateValidationTag.INVALID_DISTINGUISHED_NAMES)
-                current_score = Common.DBotScore.BAD
+        if (CertificateValidationTag.SELF_SIGNED in tags
+            and subject_dn is not None
+            and issuer_dn is not None
+                and subject_dn != issuer_dn):
```
Comment (omerKarkKatz) [inline Packs/CommonScripts/Scripts/CertificateReputation/CertificateReputation.py:224]: Why is this over indented?

File: Packs/CommonScripts/Scripts/CertificateReputation/CertificateReputation.py [hunk 4]
```diff
@@ -199,25 +201,31 @@ def dbot_context(value: str, certificate_context: Dict[str, Any]
...
             comments.append('Valid AuthorityKeyIdentifier Extension not available, some checks not performed')
 
-        if subject_key_identifier_digest is not None and authority_key_identifier_ki is not None:
-            if subject_key_identifier_digest == authority_key_identifier_ki:
-                comments.append(f'{CertificateValidationTag.SELF_SIGNED.value} Self-Signed Certificate')
-                tags.append(CertificateValidationTag.SELF_SIGNED)
-                current_score = max(current_score, Common.DBotScore.SUSPICIOUS)
+        if (subject_key_identifier_digest is not None
+            and authority_key_identifier_ki is not None
+                and subject_key_identifier_digest == authority_key_identifier_ki):
```
Comment (omerKarkKatz) [inline Packs/CommonScripts/Scripts/CertificateReputation/CertificateReputation.py:206]: Why is this over indented?

File: Packs/CommonTypes/IndicatorFields/indicatorfield-Certificate_Names.json [hunk 5]
```diff
@@ -1,7 +1,6 @@
 {
-    "associatedToAll": false,
+    "associatedToAll": true,
```
Comment (omerKarkKatz) [inline Packs/CommonTypes/IndicatorFields/indicatorfield-Certificate_Names.json:2]: Are these changes you want? 
You said you only moved to a different folder.

**Review by omerKarkKatz — COMMENTED** [review]
I looked mainly on the .py files the nested ifs looks ok but some weird indentation.

File: Packs/CommonTypes/IndicatorFields/indicatorfield-Certificate_Names.json (hunk 5 above)
Comment (Ni-Knight) [reply Packs/CommonTypes/IndicatorFields/indicatorfield-Certificate_Names.json:2]: We decided to not change it at the end, must have missed a few, fixing.

**Comment by Ni-Knight** [comment]
> I looked mainly on the .py files the nested ifs looks ok but some weird indentation.

Fixed :)

Some odd formatting.

**Comment by Ni-Knight** [comment]
Force merge is needed due to the following:
1. Lowering the script `FileCreateAndUploadV2` to version 6.0.0 to allow the test playbook to use it without breaking BC.
2. Some odd auto PEP8 that for some reason insists on a weird formatting to an if statement (Checked PEP8 documentation, it isn't correct).

**Review by AradCarmi — CHANGES_REQUESTED** [review]
Great Job!
Please update the RN of `ExpanseV2` and `X509Certificate` packs.

**Comment by Ni-Knight** [comment]
> Great Job! Please update the RN of `ExpanseV2` and `X509Certificate` packs.

Done, please go over the new RN, feel free to adjust them if needed.

**Review by AradCarmi — APPROVED** [review]

**Comment by Ni-Knight** [comment]
@ShirleyDenkberg - Can you please go over the release notes to see they are ok?

File: Packs/CommonScripts/.secrets-ignore [hunk 6]
```diff
@@ -116,4 +116,195 @@ testing@gmail.com
 a853a1b0-1ffe-4e37-d9a9-a27c6bc0bd5b@gmail.com
 cipher=ECDHE-RSA-AES128-GCM-SHA256
 07.23.05.38
-multipart/signed
\ No newline at end of file
+multipart/signed
+00:87:37:26:7f:7e:da:e8
```
Comment (altmannyarden) [inline Packs/CommonScripts/.secrets-ignore:120]: Can you check if we can remove those? As IPv6 shouldnt extract those

File: Packs/CommonTypes/IndicatorFields/indicatorfield-Validity_Not_After.json [hunk 7]
```diff
@@ -1,30 +1,28 @@
...
-    "required": false,
-    "sla": 0,
     "system": false,
-    "threshold": 72,
-    "type": "date",
+    "content": true,
+    "group": 2,
+    "hidden": false,
+    "openEnded": false,
+    "associatedToAll": true,
```
Comment (altmannyarden) [inline Packs/CommonTypes/IndicatorFields/indicatorfield-Validity_Not_After.json:21]: I dont think we should this field associated to all

File: Packs/CommonScripts/ReleaseNotes/1_13_24.md [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+##### CertificateExtract
+- Updated the Docker image to: *demisto/crypto:1.0.0.85869*.
+- New: Extract fields from a certificate file and return the standard context. (Available from Cortex XSOAR 6.0.0).
+
+
+##### New: CertificateReputation
+- New: Enrich and calculate the reputation of a certificate indicator. (Available from Cortex XSOAR 6.0.0).
+
+##### FileCreateAndUploadV2
+- Updated the Docker image to: *demisto/python3:3.10.13.85667*.
```
Comment (ShirleyDenkberg) [inline Packs/CommonScripts/ReleaseNotes/1_13_24.md:13]:
```suggestion
 New: Enrich and calculate the reputation of a certificate indicator. (Available from Cortex XSOAR 6.0.0).

##### FileCreateAndUploadV2
Updated the Docker image to: *demisto/python3:3.10.13.85667*.
```

File: Packs/CommonTypes/ReleaseNotes/3_5_0.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+#### Indicator Types
+
+- New: **X509 Certificate**
+
+#### Layouts
+
+##### New: X509 Certificate
+
+- New: New layout for the new X509 indicator type (Available from Cortex XSOAR 6.0.0).
```
Comment (ShirleyDenkberg) [inline Packs/CommonTypes/ReleaseNotes/3_5_0.md:31]:
```suggestion
New: **X509 Certificate**

#### Layouts

##### New: X509 Certificate

New: New layout for the new X509 indicator type (Available from Cortex XSOAR 6.0.0).
```

File: Packs/ExpanseV2/ReleaseNotes/1_10_49.md [hunk 10]
```diff
@@ -0,0 +1,3 @@
+#### Integrations
+##### Expanse v2
+- **Breaking Changes:** The X509 Certificate indicator type has been moved from "X509 Certificate" pack to the "Common Types" pack. Before updating please uininstall the "X509 Certificate" pack (Common Types will be updated automatically).
```
Comment (ShirleyDenkberg) [inline Packs/ExpanseV2/ReleaseNotes/1_10_49.md:3]:
```suggestion
**Breaking Changes:** The X509 Certificate indicator type has been moved from the "X509 Certificate" content pack to the "Common Types" content pack. Before updating, uninstall the "X509 Certificate" content pack (Common Types will be updated automatically).
```

File: Packs/X509Certificate/ReleaseNotes/1_0_37.md [hunk 11]
```diff
@@ -0,0 +1,10 @@
+
+- **X509 Certificate**
+**Breaking Changes:** The X509 Certificate indicator type along with all its relevant fields has been moved from "X509 Certificate" pack to the "Common Types" pack. Before updating please uninstall the "X509 Certificate".
```
Comment (ShirleyDenkberg) [inline Packs/X509Certificate/ReleaseNotes/1_0_37.md:3]:
```suggestion
**Breaking Changes:** The X509 Certificate indicator type along with all its relevant fields has been moved from the "X509 Certificate" content pack to the "Common Types" content pack. Before updating, uninstall the "X509 Certificate" content pack.
```

File: Packs/X509Certificate/ReleaseNotes/1_0_37.md [hunk 12]
```diff
@@ -0,0 +1,10 @@
...
+
+#### Scripts
+##### CertificateExtract
+- **Breaking Changes:** The script has been moved to Common Scripts
+
+##### CertificateReputation
+- **Breaking Changes:** The script has been moved to Common Scripts
```
Comment (ShirleyDenkberg) [inline Packs/X509Certificate/ReleaseNotes/1_0_37.md:10]:
```suggestion
**Breaking Changes:** The script has been moved to Common Scripts.

##### CertificateReputation
**Breaking Changes:** The script has been moved to Common Scripts.
```

**Comment by ShirleyDenkberg** [comment]
@altmannyarden @AradCarmi @omerKarkKatz Doc review completed.

File: Packs/CommonScripts/.secrets-ignore (hunk 6 above)
Comment (Ni-Knight) [reply Packs/CommonScripts/.secrets-ignore:120]: Those are there due to the test data, not IPv6

**Comment by Ni-Knight** [comment]
@ShahafBenYakir ready for force merge.
Force is needed as we lowered the support for one of the scripts to 6.0.0 so the playbook which uses it will have it available. The playbook was using a now deprecated script and that is the root cause for the change.

**Pull request merged**